tests/rbac_test.go

```diff
@@ -0,0 +1,212 @@
+package tests_test
+
+import (
+	"flag"
+	"fmt"
+	"io"
+	"strings"
+	"time"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/ginkgo/extensions/table"
+	. "github.com/onsi/gomega"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	ktests "kubevirt.io/kubevirt-ansible/tests"
+	"kubevirt.io/kubevirt/pkg/kubecli"
+	"kubevirt.io/kubevirt/pkg/log"
+	"kubevirt.io/kubevirt/tests"
+)
+
+var _ = Describe("RBAC", func() {
+	flag.Parse()
+
+	var vmi ktests.VMManifest
+
+	virtClient, err := kubecli.GetKubevirtClient()
+	tests.PanicOnError(err)
+
+	BeforeEach(func() {
+		By("Login with the user test_admin")
+		args := []string{"login", "-u", "test_admin", "-p", "123456"}
+		_, err := tests.RunCommand("oc", args...)
+		Expect(err).ToNot(HaveOccurred())
+
+		tests.BeforeTestCleanup()
+		vmi.Name = "vm-cirros"
+		vmi.Manifest = "tests/manifests/vm-cirros.yaml"
+		ktests.CreateUser()
+
+		By("Create a VM with yaml")
```

Comment:
@adityaramteke I don't understand the creation of VM/VMI in `BeforeEach`, why is it created here when it is tested in in the `DescribeTable`?

Reply:
@vatsalparekh I need to set up the environment for testing `view` role. I'm creating VM/VMI within the namespace `kubevirt-test-alternative`. The user having `view` role can not able to create/modify VM/VMI and can not able to access the subresource endpoint. Those changes are not going to affect for the admin/edit role that I'm testing in `DescribeTable` because these will be tested within the different namespace `kubevirt-test-user`.

Reply:
@adityaramteke so this means you are creating Vms for the other 2 roles as well when it is not required. I would suggest to have a func separately maybe, and in your `DescribeTable` only when the `view` role is passed, then it creates the VMs in alternative namespace.
Also, this seems confusing at first, so please add some comments there.

Reply:
done